src/sqlfluff/dialects/dialect_clickhouse.py

```diff
@@ -564,12 +564,42 @@ class CreateMaterializedViewStatementSegment(BaseSegment):
     )
 
 
+class CreateDatabaseStatementSegment(ansi.CreateDatabaseStatementSegment):
+    """A `CREATE DATABASE` statement.
+
+    As specified in
+    https://clickhouse.com/docs/en/sql-reference/statements/create/database
+    """
+
+    type = "create_database_statement"
+
+    match_grammar = Sequence(
+        "CREATE",
+        "DATABASE",
+        Ref("IfNotExistsGrammar", optional=True),
+        Ref("DatabaseReferenceSegment"),
+        Sequence(
+            "ON",
+            "CLUSTER",
+            Ref("ExpressionSegment"),
+            optional=True,
+        ),
+        Ref("EngineSegment"),
```

Comment:
1. I don't think you need to use an engine for tables
2. It is also optional

Reply:
Good point!

PR still in WIP.